lib/Controller/RoomController.php

```diff
@@ -1103,6 +1104,16 @@ public function addParticipantToRoom(string $newParticipant, string $source = 'u
 		/** @var IUser $addedBy */
 		$addedBy = $this->userManager->get($this->userId);
 
+		//Check if the newParticipant is banned
+		try {
+			$this->banService->throwIfActorIsBanned($this->room, $newParticipant);
+		} catch (ForbiddenException $e) {
+			$this->logger->info('Participant ' . ($newParticipant ?? 'guest') . ' is banned from room ' . $this->room->getToken() . ' by ' . $e->getMessage());
+			return new DataResponse([
+				'error' => 'ban',
+			], Http::STATUS_FORBIDDEN);
+		}
```

Comment:
Wrap in `if ($source === 'users') {`

Guests and IPs can't be added, and nothing else can be banned.